### DISCUSSION TITLE
Querying for multi-match with `&&`

### DISCUSSION THREAD

**Discussion by VenatusSimpleX** [pocketbase/pocketbase#2574] [Q&A] [answered]
Hi there! I'm wondering if I am missing something when trying to query for a multi-match, so basically, if I do a match like below, it works:
<img width="849" alt="image" src="https://github.com/pocketbase/pocketbase/assets/31540342/628b10d1-0212-4612-a17f-786f28a6d407">

But, the moment I try to do a multi-match, no results are returned:
<img width="967" alt="image" src="https://github.com/pocketbase/pocketbase/assets/31540342/6e36cb2e-f9ee-49d7-8916-1570b4552744">

Based on what I understand, this should at the very least return the `Seaside Retreat Tank` product, because both `any`/`?=` queries matches with the product, am I misunderstanding something here?

**Comment by ganigeorgiev** [comment, marked as answer]
I think this is similar to https://github.com/pocketbase/pocketbase/discussions/2367.

In your case `categories.name` refers to a single `categories` row and it can't be 2 things at the same time.

Internally the above is converted to a SQL query like:

```sql
-- simplified pseudo SQL
SELECT DISTINCT example.*
FROM example
LEFT JOIN brand
LEFT JOIN categories
WHERE brand.id LIKE "%%" AND (
    categories.name = "Shorts" AND categories.name = "Lingerie"
)
```

_`IN`-like operator support will be added in the near future._

**Reply by VenatusSimpleX** [reply]
Thanks for the direction! We really need a text-based search (as we may have same name but different ids), so I think we'll make a view-only table, concat the category names, then do a text search there.
